parser.mly

```diff
@@ -145,8 +151,9 @@ instruction:
   label=label
   LBRACKET cond=separated_list(COMMA, expression) RBRACKET
   LPAREN func=label COMMA version=label COMMA pos=label RPAREN
-  LBRACKET map=separated_list(COMMA, osr_def) RBRACKET
-  { Osr {label; cond; target= {func; version; pos}; map} }
+  top=osr_map
+  frames=separated_list(COMMA, osr_frame)
+  { Osr {label; cond; target= {func; version; pos}; varmap=top; frame_maps=frames} }
```

Comment:
I would rather like the first frame to also be separated from the osr_map by a comma. You can do this easily (and this can simplify `osr_frame` above as well) by just using `list(preceded(COMMA, foo))` instead of `separated_list(COMMA, foo)`. You can also give this pattern a name:

```
%inline preparated_list(sep, elem): list(preceded(sep, elem)) { $1 }
```